src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/BackedEnumValueResolver.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller\ArgumentResolver;
+
+use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+
+/**
+ * Attempt to resolve backed enum cases from request attributes.
+ *
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ */
+class BackedEnumValueResolver implements ArgumentValueResolverInterface
+{
+    public function supports(Request $request, ArgumentMetadata $argument): bool
+    {
+        if (!is_subclass_of($argument->getType(), \BackedEnum::class, true)) {
+            return false;
+        }
+
+        // do not support if no value can be resolved at all
+        // letting the \Symfony\Component\HttpKernel\Controller\ArgumentResolver\DefaultValueResolver be used
+        // or \Symfony\Component\HttpKernel\Controller\ArgumentResolver fail with a meaningful error.
+        return $request->attributes->has($argument->getName());
+    }
+
+    public function resolve(Request $request, ArgumentMetadata $argument): iterable
+    {
+        /** @var interface-string<\BackedEnum> $enumType */
+        $enumType = $argument->getType();
+        $values = $argument->isVariadic() ? $request->attributes->all($argument->getName()) : [$request->attributes->get($argument->getName())];
+
+        foreach ($values as $value) {
+            if (null === $value) {
+                yield null;
+
+                continue;
+            }
+
+            try {
+                yield $enumType::from($value);
+
+                continue;
+            } catch (\ValueError|\TypeError $error) {
+                throw new BadRequestException(sprintf('Could not resolve the "%s" controller argument: %s', $argument->getName(), $error->getMessage()), $error->getCode(), $error);
```

Comment:
I would expect a failed hydration of the enum to result in a 404 error.

Reply:
That's arguable. But for the main use-case targeted by this PR, i.e route parameters, it would make sense indeed.
Let's assume `\ValueError` => not found?

Since variadic would only be possible by using attributes created in the request by other means (listeners, route defaults, …), we might want to cut down the use-cases for this core resolver as well, and eventually drop out support for variadic? 
Hence any other possible issue would be a logic exception.

Reply:
👍🏻 I don't see a use-case for variadics either at the moment.